03_regex_email_ip_url.rb

```diff
@@ -0,0 +1,48 @@
+# Write Regex for email phone_number IP URL
+# Read about each_with_index, map(), with_index, select, collect
+
+email = "rohit@joshi.com"
+mobile = "8554801616"
+
+if email.match(/^\w+@\w+\.com$/)
+  puts "valid email" 
+else 
+  puts "invalid email"
+end
```

Comment:
Can you move this to a method that returns a boolean value and then based on that value we can show the message. 
By doing this way we can reuse the method.

Reply:
Sure, how about creating a single method, say `regex_matcher(string, type)` which accepts the string to be matched, checks if its `nil` or not and depending upon the type passed (email, ip, URL) validates using different regex.

Reply:
@rohitjoshixyz Great approach. But one possible downside is regex_matcher is carrying too much responsibility. When we have small methods `valid_email?` etc. You can now individually use this method elsewhere.

Reply:
Okay so it is advisable to go for smaller and specific (hence more readable) methods rather than a general but flexible method in Ruby. Correct?

Reply:
Smaller method are preferable as they are composable. We can reuse the smaller method in regex_matcher and don't have to repeat the code.